src/contribute/index.md

```diff
@@ -1,26 +1,183 @@
+the terms under which intellectual property has been contributed to the ASF and
+thereby allow us to defend the project should there be a legal dispute
+regarding the software at some future time.
+
+If you are contributing a `PTransform` to Beam, we have an extensive
+[PTransform Style Guide]({{ site.baseurl }}/contribute/ptransform-style-guide).
+
+### Building & Testing
+
+We use Gradle to orchestrate building and testing.
+
+The entire set of tests can be run with this command at the root of the git
+repository.
+
+    $ ./gradlew check
+
+You can limit testing to a particular module and Gradle will build just the
+necessary things to run those tests. For example:
+
+    $ ./gradlew -p sdks/go check
+    $ ./gradlew -p sdks/java/io/cassandra check
+    $ ./gradlew -p runners/flink check
+
+### Testing the Python SDK
+
+You can directly use the Python toolchain instead of having Gradle orchestrate
+it. This may be faster for you. We recommend setting up a virtual environment
+before testing your code.
+
+If you update any of the [cythonized](http://cython.org) files in Python SDK,
+you must install the `cython` package before running following command to
+properly test your code. 
+
+The following commands should be run in the `sdks/python` directory.
+This command runs all Python tests.
+
+    $ python setup.py nosetests
+
+You can use following command to run a single test method.
+
+    $ python setup.py nosetests --tests <module>:<test class>.<test method>
+
+    Example:
+    $ python setup.py nosetests --tests apache_beam.io.textio_test:TextSourceTest.test_progress
+
+To check just for lint errors, run the following command.
+
+    $ ../../gradlew lint
+
+#### Remote testing
+
+This step is only required for testing SDK code changes remotely (not using
+directrunner). In order to do this you must build the Beam tarball. From the
+root of the git repository, run:
+
+```
+$ cd sdks/python/
+$ python setup.py sdist
+```
+
+Pass the `--sdk_location` flag to use the newly built version. For example:
+
+```
+$ python setup.py sdist > /dev/null && \
+    python -m apache_beam.examples.wordcount ... \
+        --sdk_location dist/apache-beam-2.5.0.dev0.tar.gz
+```
+
+## Contributing to the website
+
+The Beam website is in the [Beam Site GitHub
+mirror](https://github.com/apache/beam-site) repository in the `asf-site`
+branch (_not_ `master`).  The
+[README](https://github.com/apache/beam-site/blob/asf-site/README.md) there
+explains how to modify different parts of the site. The GitHub workflow is the
+same - make your change and open a pull request.
+
+Issues are tracked in the
+[website](https://issues.apache.org/jira/issues/?jql=project%20%3D%20BEAM%20AND%20component%20%3D%20website)
+component in JIRA.
+
+## Works in progress
+
+A great way to contribute is to join an existing effort. There are many
+works in progress, some on branches because they are very incomplete.
+
+### Portability Framework
```

Comment:
@herohde @bsidhom This is a fairly abbreviated link, but I thought it was the most important work-in-progress to highlight. It might be nice to break it out into efforts for each SDK and runner. Or maybe that should be just part of the linked doc. Anyhow, take a look at http://apache-beam-website-pull-requests.storage.googleapis.com/438/contribute/index.html#works-in-progress and see what you think.

Reply:
I think mentioning the Flink runner more prominently both here and on the portability page would be great. I also wanted to reference the progress sheet under status, so it might be a good context to call out Flink.

I don't think we should make it a new section, because it's not a new runner in the same way the others are. But I don't feel very strongly about it.

Reply:
Could I prevail upon one of you two to follow up this PR or issue a PR to my PR?